### PR TITLE
DEVX-1040 Enable typescript strict mode

### PR DESCRIPTION
**Pull request by pajcho**
This PR introduces a strict typescript mode and fixes all type issues. Alongside with that it introduces some improvements to our linting process, fixes some typos and some general improvements.

⚠️ Since the strict types are kind of a breaking change, this will require a minor release version bump.

### CODE REVIEW

File: examples/allowAndTransfer/index.ts [hunk 1]
```diff
@@ -82,6 +88,6 @@ const myTokenAbiInterface = new Interface(myTokenAbi);
       process.exit(1);
     }
 
-    console.log(e.response);
+    console.log((e as { response: unknown }).response);
```
Comment (pajcho) [inline examples/allowAndTransfer/index.ts:91]: Simplify to just `console.log(e)`

File: lib/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+type TenderlyEnvVariables = {
+  TENDERLY_ACCESS_KEY: string;
+  TENDERLY_ACCOUNT_NAME: string;
+  TENDERLY_PROJECT_NAME: string;
+  TENDERLY_GET_BY_PROJECT_NAME: string;
+};
+
+export function getEnvironmentVariables() {
+  return process.env as TenderlyEnvVariables;
+}
```
Comment (pajcho) [inline lib/helpers.ts:10]: @maleksandar to think about how to organise this

File: lib/repositories/Repository.ts [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+
 export interface Repository<T> {
   get: (uniqueId: string) => Promise<T | undefined>;
   add: (uniqueId: string, data: Partial<T>) => Promise<T | undefined>;
-  remove: (uniqueId) => Promise<void>;
-  update: <UpdateableFields>(uniqueId, data: UpdateableFields) => Promise<unknown | undefined>;
-  getBy: (queryObject: unknown) => Promise<T[] | undefined>;
+  remove: (uniqueId: string) => Promise<void>;
+  update: (uniqueId: string, data: never) => Promise<unknown | undefined>;
+  getBy: (queryObject: GetByParams) => Promise<T[] | undefined>;
```
Comment (pajcho) [inline lib/repositories/Repository.ts:8]: replace `GetByParams` with never or other generic type

**Comment by IgorPetkovic** [comment]
> ⚠️ Since the strict types are kind of a breaking change, this will require a minor release version bump.
Can you elaborate a bit on "kind of a breaking change"? If this will require changes by existing users, i.e. their current setup will break, it needs to be a major version bump in that case.
See https://semver.org/#summary

File: lib/helpers.ts (hunk 2 above)
Comment (maleksandar) [reply lib/helpers.ts:10]: for now, here is ok. We don't have many exports from the main path, once we have such an issue, we might introduce namespaces with a major release.

**Review by maleksandar — APPROVED** [review]

**Pull request merged**